### PR TITLE
Remove #define's adding libreport_ to function names

### PR DESCRIPTION
**Pull request by michalfabik**
This PR is analogous to https://github.com/abrt/abrt/pull/1466. Mass-rename functions and globals to `libreport_\1` where `#define`s have been used.

### CODE REVIEW

File: src/client-python/reportclient/clientmodule.c [hunk 1]
```diff
@@ -23,12 +23,12 @@
...
-    { "ask_password"              , p_ask_password            , METH_VARARGS },
-    { "ask_yes_no"                , p_ask_yes_no              , METH_VARARGS },
-    { "ask_yes_no_yesforever"     , p_ask_yes_no_yesforever   , METH_VARARGS },
-    { "ask_yes_no_save_result"    , p_ask_yes_no_save_result  , METH_VARARGS },
+    { "libreport_alert"                     , p_alert                   , METH_VARARGS },
+    { "libreport_ask"                       , p_ask                     , METH_VARARGS },
+    { "libreport_ask_password"              , p_ask_password            , METH_VARARGS },
+    { "libreport_ask_yes_no"                , p_ask_yes_no              , METH_VARARGS },
+    { "libreport_ask_yes_no_yesforever"     , p_ask_yes_no_yesforever   , METH_VARARGS },
+    { "libreport_ask_yes_no_save_result"    , p_ask_yes_no_save_result  , METH_VARARGS },
```
Comment (ernestask) [inline src/client-python/reportclient/clientmodule.c:31]: Err, I don’t think this is correct.

**Comment by ernestask** [comment]
Can you also go through dependent packages in Fedora and see how much fallout we’re looking at?

File: src/client-python/reportclient/clientmodule.c (hunk 1 above)
Comment (michalfabik) [reply src/client-python/reportclient/clientmodule.c]: Reverted.

**Comment by michalfabik** [comment]
I've found the following with `dnf repoquery --whatdepends`:  

anaconda-core    
blivet-gui-runtime    
dnf-data    
initial-setup
mdadm    
python3-meh    
python3-meh-gui
reportd
rubygem-abrt
setroubleshoot

**Comment by ernestask** [comment]
I’ll update your comment with details. Some of those only depend on installed configs or workflows.

**Comment by ernestask** [comment]
Okay, so only reportd seems to be an issue and it’s our project. Should be a trivial task.

File: src/include/ureport.h [hunk 2]
```diff
@@ -94,9 +92,8 @@ ureport_server_config_destroy(struct ureport_server_config *config);
  *
  * @param config a server configuration to be populated
  */
-#define ureport_server_config_load libreport_ureport_server_config_load
 void
-ureport_server_config_load(struct ureport_server_config *config,
+libreport_ureport_server_config_load(struct ureport_server_config *config,
                            map_string_t *settings);
```
Comment (ernestask) [inline src/include/ureport.h:97]: The broken alignment will be an issue everywhere. Can be addressed later ad-hoc.

File: src/include/libreport_types.h [hunk 3]
```diff
@@ -25,90 +25,67 @@
 typedef gchar **string_vector_ptr_t;
 typedef const gchar *const *const_string_vector_const_ptr_t;
 
-#define string_vector_new_from_string libreport_string_vector_new_from_string
-string_vector_ptr_t string_vector_new_from_string(const char *vector);
-#define string_vector_free libreport_string_vector_free
-void string_vector_free(string_vector_ptr_t vector);
+string_vector_ptr_t libreport_string_vector_new_from_string(const char *vector);
```
Comment (ernestask) [inline src/include/libreport_types.h:28]: Unrelated, but all of this code needs to eventually be shot into the Sun, so no trace remains.

File: src/cli/cli-report.c [hunk 4]
```diff
@@ -230,18 +230,18 @@ static int read_crash_report_field(const char *text, problem_data_t *report,
     char newvalue[length + 1];
     strncpy(newvalue, textfield, length);
     newvalue[length] = '\0';
-    strtrim(newvalue);
+    libreport_strtrim(newvalue);
```
Comment (ernestask) [inline src/cli/cli-report.c:233]: Note: this crap can be replaced with `g_strchug()` and `g_strchomp()` where needed.

Comment (ernestask) [reply src/cli/cli-report.c:233]: Part of the future “nuke strbuf” quest.

File: src/cli/cli-report.c [hunk 5]
```diff
@@ -711,11 +711,11 @@ static int choose_number_from_range(unsigned min, unsigned max, const char *mess
     {
         char *answer = libreport_ask(message);
 
-        picked = xatou(answer);
+        picked = libreport_xatou(answer);
```
Comment (ernestask) [inline src/cli/cli-report.c:714]: Unrelated, but the code for this is also a whole bunch of “what”.

File: src/cli/cli-report.c [hunk 6]
```diff
@@ -743,14 +743,14 @@ static char *select_event_name(GList *list_options)
...
-    return xstrdup((char*)chosen->data);
+    return libreport_xstrdup((char*)chosen->data);
 }
 
 int select_and_run_one_event(const char *dump_dir_name, const char *pfx, int interactive)
 {
     GList *list_events = list_possible_events_glist(dump_dir_name, pfx);
     char *event_name = select_event_name(list_events);
-    list_free_with_free(list_events);
+    libreport_list_free_with_free(list_events);
```
Comment (ernestask) [inline src/cli/cli-report.c:753]: Unrelated: these can be replaced with `g_autolist` pointers.

File: src/gui-wizard-gtk/wizard.c [hunk 7]
```diff
@@ -1966,7 +1966,7 @@ static bool check_minimal_bt_rating(const char  *event_name,
 
     if (!event_name)
         error_msg_and_die(_("Cannot check backtrace rating because of invalid event name"));
-    else if (prefixcmp(event_name, "report") != 0)
+    else if (libreport_prefixcmp(event_name, "report") != 0)
```
Comment (ernestask) [inline src/gui-wizard-gtk/wizard.c:1969]: `g_str_has_prefix()` also exists already.

**Pull request merged**